slatedb/src/manifest/store.rs

```diff
@@ -80,115 +75,116 @@ impl FenceableManifest {
         manifest_update_timeout: Duration,
         system_clock: Arc<dyn SystemClock>,
     ) -> Result<Self, SlateDBError> {
-        Self::init(
-            stored_manifest,
-            |m| m.writer_epoch,
-            |m, e| m.writer_epoch = e,
+        // Initialize generic fenceable record using writer epoch
+        let fr = FenceableRecord::init(
+            stored_manifest.inner,
             manifest_update_timeout,
             system_clock,
+            |m: &Manifest| m.writer_epoch,
+            |m: &mut Manifest, e: u64| m.writer_epoch = e,
         )
-        .await
+        .await?;
+        let local_epoch = fr.local_epoch;
+        Ok(Self {
```

Comment:
Why do we need the `local_epoch` and `stored_epoch` here. We can also just make `FenceableManifest` a thin wrapper around `FenceableRecord<Manifest>`